libs/plugins/modifiercompiler.unescape.php

```diff
@@ -39,7 +39,7 @@ function smarty_modifiercompiler_unescape($params, Smarty_Internal_TemplateCompi
         case 'entity':
         case 'htmlall':
             if (Smarty::$_MBSTRING) {
-                return 'mb_convert_encoding(' . $params[ 0 ] . ', ' . $params[ 2 ] . ', \'HTML-ENTITIES\')';
+                return 'html_entity_decode(mb_convert_encoding(' . $params[ 0 ] . ', ' . $params[ 2 ] . ', \'UTF-8\'), ENT_QUOTES | ENT_SUBSTITUTE | ENT_HTML401, ' . $params[ 2 ] . ')';
```

Comment:
I was triggered by the difference between `ENT_QUOTES | ENT_SUBSTITUTE | ENT_HTML401` for multibyte and `ENT_NOQUOTES` for no multibyte. Seemed off. The unit tests didn't test for quotes, so I added them. Turns out your change is better, so I also ported it to line 44. Agree?

Reply:
@Progi1984 any thoughts on my last commit here?

Reply:
Just checked and I approved. Really good job.